test/unit/logging.test.ts

```diff
@@ -0,0 +1,323 @@
+import sinon from 'sinon';
+
+let { timeout, mockApis, wss, ws, closeWebSocketServer } = require('../test-utils');
+
+const { log } = require('../../src/logging');
+const sandbox = sinon.createSandbox();
+
+describe('Logging', () => {
+
+  beforeAll(async () => {
+    await closeWebSocketServer();
+  });
+  afterAll(async () => {
+    await closeWebSocketServer();
+  });
+
+  afterEach(async () => {
+    if (ws) {
+      await Promise.resolve(ws.close());
+      ws = null;
+    }
+    if (wss) {
+      wss.removeAllListeners();
+    }
+    sandbox.restore();
+  });
+
+  test('_log | will not notify logs if the logLevel is lower than configured', async () => {
+    const { sdk } = mockApis({ withLogs: true });
+    sdk._logLevel = 'warn';
+    await sdk.initialize();
+    sinon.stub(sdk._backoff, 'backoff'); // called in notifyLogs
```

Comment:
You can use jest's mocking library instead here, and eliminate sinon entirely.
`jest.spyOn(sdk._backoff, 'backoff')`